### PR TITLE
Add method handle related hidden fields to JITServer VM info

### PR DESCRIPTION
**Pull request by dmitry-ten**
- The new method handle transformer uses hidden fields
`javaVM->vmtargetOffset` and `javaVM->vmindexOffset`.
These offsets are set by the client VM, and should be accessed
via a front-end query.
- This commit adds front-end queries to access the aforementioned fields
correctly on JITServer and cache them in client session's VM info.

### CODE REVIEW

**Comment by dmitry-ten** [comment]
@mpirvu ready for review

**Comment by dmitry-ten** [comment]
@nbhuiyan fyi.

**Review by mpirvu — APPROVED** [review]
LGTM

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit jdk11,jdk17

**Comment by mpirvu** [comment]
There are some conflicts that need to be resolved. I think this happened after I merged #13408

**Comment by mpirvu** [comment]
Looking at PPC, there are more failures than usual. I cannot tell if this PR caused the extra failures or we are talking about intermittent failures.

**Review by mpirvu — APPROVED** [review]

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit jdk11,jdk17

**Comment by dmitry-ten** [comment]
I see lots of segfaults on power. Will investigate if it's related to this change.

**Comment by mpirvu** [comment]
There are 48 failures with Test_openjdk17_j9_sanity.functional_x86-64_linux_jit_Personal.
It may be that the code as it stands does not include the fix from https://github.com/eclipse-openj9/openj9/pull/13425
@dmitry-ten Could you please rebase if that's the case? Thanks

**Comment by dmitry-ten** [comment]
Oh, true. Rebased.

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit jdk17

**Comment by mpirvu** [comment]
Something is off. There are 49 test failures on x86 and 26 on Power. Could you please test this PR code locally? The code looks fine as far as I can tell, but I am worried by the large number of failures. Thanks

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit jdk11

**Comment by dmitry-ten** [comment]
@mpirvu the problem was that `vmtargetOffset` and `vmindexOffset` will not change once they are set, but they are set at a later point.
So when we cache VM info, the offsets haven't been set yet, and will be 0.
The solution is to add messages to fetch the offsets from the client when they are guaranteed to be set, and cache them to VM info at that point.
The failing tests should pass now.

File: runtime/compiler/env/VMJ9Server.cpp [hunk 1]
```diff
@@ -2257,6 +2257,32 @@ TR_J9ServerVM::delegatingMethodHandleTargetHelper(TR::Compilation *comp, TR::Kno
    knot->updateKnownObjectTableAtServer(idx, std::get<1>(recv));
    return idx;
    }
+
+UDATA
+TR_J9ServerVM::getVMTargetOffset()
+   {
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   auto *vmInfo = _compInfoPT->getClientData()->getOrCacheVMInfo(stream);
+   if (vmInfo->_vmtargetOffset)
```
Comment (mpirvu) [inline runtime/compiler/env/VMJ9Server.cpp:2266]: is it guaranteed that a value of 0 is not valid?

File: runtime/compiler/control/JITClientCompilationThread.cpp [hunk 2]
```diff
@@ -463,6 +463,11 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
                vmInfo._srConstructorAccessorClass = NULL;
 #endif // J9VM_OPT_SIDECAR
          vmInfo._extendedRuntimeFlags2 = javaVM->extendedRuntimeFlags2;
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+         // These offsets are initialized later on
+         vmInfo._vmtargetOffset = 0;
```
Comment (mpirvu) [inline runtime/compiler/control/JITClientCompilationThread.cpp:468]: Isn't better to read the values from the VM? We could save a message if those values are not 0 when we read them.

**Review by mpirvu — APPROVED** [review]
I have a couple of questions, but otherwise it looks good.

File: runtime/compiler/env/VMJ9Server.cpp (hunk 1 above)
Comment (dmitry-ten) [reply runtime/compiler/env/VMJ9Server.cpp:2266]: I do not know if it is guaranteed, but I've never seen a valid value be 0.
In the worst-case scenario, if the valid value is 0, we'll just end up sending a message every time this query is called

File: runtime/compiler/control/JITClientCompilationThread.cpp (hunk 2 above)
Comment (dmitry-ten) [reply runtime/compiler/control/JITClientCompilationThread.cpp:468]: I've never seen the VM query return anything but 0 at this point, so I just decided to set it to 0, to indicate that the value is not valid yet.

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit jdk17

**Comment by mpirvu** [comment]
@dmitry-ten please open a PR for 0.28.0 branch.

**Pull request merged**